src/interfaces/providers.js

```diff
@@ -168,6 +175,18 @@ module.exports = function (app) {
   }
 }
 
+function isNumber(s) {
```

Comment:
My brain hurts parsing this.

We are going to be feeding the value anyway to `parseInt`,  that does not throw errors and handles stuff like removing whitespace. So what if we simply call parseInt and check the return value for bounds? No separate isNumber precheck.

Reply:
I have updated logic to use just `parseInt` function. Now it is much cleaner